packages/flutter/test/widgets/nested_scroll_view_test.dart

```diff
@@ -212,6 +212,65 @@ void main() {
     expect(context.clipBehavior, equals(Clip.antiAlias));
   });
 
+  testWidgets('NestedScrollView always scrolls outer scrollable first, (double precision errors, #136199)', (WidgetTester tester) async {
+    final Key innerKey = UniqueKey();
+    final Key outerKey = UniqueKey();
+
+    final ScrollController outerController = ScrollController();
+
+    Widget build() {
+      return Directionality(
+        textDirection: TextDirection.ltr,
+        child: Scaffold(
+          body: NestedScrollView(
+            key: outerKey,
+            controller: outerController,
+            physics: const BouncingScrollPhysics(),
+            headerSliverBuilder: (BuildContext context, bool innerBoxIsScrolled) => <Widget>[
+              SliverToBoxAdapter(
+                child: Container(color: Colors.green, height: 300),
+              ),
+              SliverOverlapAbsorber(
+                handle: NestedScrollView.sliverOverlapAbsorberHandleFor(context),
+                sliver: SliverToBoxAdapter(
+                  child: Container(
+                    color: Colors.blue,
+                    height: 64,
+                  ),
+                ),
+              ),
+            ],
+            body: SingleChildScrollView(
+              key: innerKey,
+              physics: const BouncingScrollPhysics(),
+              child: Container(
+                decoration: const BoxDecoration(
+                  gradient: LinearGradient(
+                    begin: Alignment.topCenter,
+                    end: Alignment.bottomCenter,
+                    colors: <ui.Color>[Colors.black, Colors.blue],
+                    stops: <double>[0, 1],
+                  ),
+                ),
+                height: 800,
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(build());
+
+    outerController.addListener(() {
+      assert(false); // We do not want this to be called.
```

Comment:
Why? The outer should scroll before the inner, why would the outer not scroll?

Reply:
The way I understand the test is that we have:
`tester.drag(find.byKey(innerKey), const Offset(0, 2000));`

Which means dragging the finger down the screen (I believe). So we are immediately overscrolling and triggering the dampening effect of the spring. The total height is 300 + 64 + 800 from the individual components. 1164 < 2000. So we have technically 836 units to work with.

`tester.fling(find.byKey(innerKey), const Offset(0, -600), 2000);`

After that we immediately fling the finger up the screen. My understanding is that the total distance will be 600 units up and the 2000 just affects how quickly the 600 will happen.

Which equals 836 - 600 = 236. So we still have 236 units to spare before the outer scrollable should start scrolling. What does not fly is the spring does not actually "stretch" to 836 visually, but I have no idea how that is implemented. If this description makes sense I will add it to the test itself, but there is a possibility it is nonsense and I just do not understand what I wrote ;).

Reply:
Hmm, it may be easier to understand if you add checks on what the actual position is before and after each gesture. What is the value of position.pixels for the outer and inner each time? For the nested scroll view in general, the inner should not scroll before the outer has fully scrolled to its end.

If we are dragging down first, overscrolling the outer, then flinging up, I would expect the outer to consume some of the distance flinging up before the inner receives any.

Reply:
> the outer is not being overscrolled

Ah right I see. The inner is using bouncing scroll physics. Yup. So the inner overscrolls at the top, but then when the test flings the inner up, I expect the outer to consume some of the distance before the inner does.

> Since we are not supposed to assign a scroll controller to the inner scrollview I do not have output from that.

The inner scroll controller is accessible through the inner key you have created in the test.

Can you add checks that validate the inner and outer positions at each step to the test?

Reply:
Alright, I added more asserts to the test, I did not find a way to obtain the controller from the previous setup so I just borrowed someone else's solution :). The outer scrollable does not move, it may be surprising but it's true (according to the `expect()` and the `.addListener()`)